Server/App/Dispatcher/TopicDispatcher.php

```diff
@@ -130,11 +131,30 @@ public function onPublish(ConnectionInterface $conn, Topic $topic, WampRequest $
     public function dispatch($calledMethod, ConnectionInterface $conn = null, Topic $topic, WampRequest $request, $payload = null, $exclude = null, $eligible = null, $provider = null)
     {
         $dispatched = false;
-
+        
         if ($topic) {
+
             foreach ((array) $request->getRoute()->getCallback() as $callback) {
                 $appTopic = $this->topicRegistry->getTopic($callback);
 
+                if ( $appTopic instanceof FirewalledTopicInterface )
+                {
+                    $error = $appTopic->checkConnection($conn, $topic, $request, $payload , $exclude , $eligible , $provider );
```

Comment:
As said previously, we do not have to check the return result, just try catch, if catch, a firewall exception, close the connection, otherwise, it's ok